esmvalcore/dataset.py

```diff
@@ -848,3 +849,21 @@ def _update_timerange(self):
         check.valid_time_selection(timerange)
 
         self.set_facet('timerange', timerange)
+
+    def get_temporary_fixed_file_dir(self) -> Path:
+        """Create and return new temporary directory for storing fixed files.
+
+        Returns
+        -------
+        Path
+            Path to new temporary directory.
+
+        """
+        fixed_file_dir = self.session.fixed_file_dir
+        fixed_file_dir.mkdir(parents=True, exist_ok=True)
+        facets_for_prefix = ('project', 'dataset', 'mip', 'short_name')
+        prefix = '_'.join(
+            [str(self.facets.get(facet, '')) for facet in facets_for_prefix] +
```

Comment:
```suggestion
            ["-".join(str(elem) for elem in value) if isinstance(value, (list, tuple)) else str(value) for facet, value in self.facets.items() if facets in facets_for_prefix] +
```
would be nicer since it avoids sequences of underscores and the characeters `[](),` in the directory name (the latter only when we have the full list of facets from the `summary` method).